src/toil/jobStores/aws/jobStore.py

```diff
@@ -1047,3 +1011,75 @@ def toItem(self):
         :return: a str for the item's name and a dictionary for the item's attributes
         """
         return self.jobStoreID, self.binaryToAttributes(cPickle.dumps(self))
+
+
+def copyKeyMultipart(srcKey, dstBucketName, dstKeyName, headers=None):
+    """
+    Copies a key from a source key to a destination key in multiple parts. Note that if the
+    destination key exists it will be overwritten implicitly, and if it does not exist a new
+    key will be created. If the destination bucket does not exist an error will be raised.
+
+    :param boto.s3.key.Key srcKey: The source key to be copied from.
+    :param str dstBucketName: The name of the destination bucket for the copy.
+    :param str dstKeyName: The name of the destination key that will be created or overwritten.
+    :param dict headers: Any headers that should be passed.
+
+    :rtype: boto.s3.multipart.CompletedMultiPartUpload
+    :return: An object representing the completed upload.
+    """
+    s3 = boto.connect_s3()
+    partSize = defaultPartSize
+    totalSize = srcKey.size
+    totalParts = (totalSize / partSize) + 1 if totalSize % partSize else (totalSize / partSize)
+    exceptionList = []
+
+    def _copyPart(partIndex):
+        if exceptionList:
+            return None
+        else:
+            try:
+                startByte = partIndex * partSize
+                startByte = startByte if startByte <= totalSize else totalSize - ((partIndex - 1) * partSize)
+
+                part = upload.copy_part_from_key(src_bucket_name=srcKey.bucket.name,
+                                                 src_key_name=srcKey.name,
+                                                 part_num=partIndex + 1,  # Part numbers are indexed from 1 in AWS.
+                                                 start=startByte,
+                                                 end=min(startByte + partSize, totalSize) - 1,
+                                                 headers=headers or {})
+
+                log.debug('Copied part %d of %d' % (partIndex, totalParts))
+
+            except:
+                exceptionList.append({'partIndex': partIndex, 'tb': traceback.format_exc()})
```

Comment:
Change of plan. Since we only log the first four backtraces, use `format_exc` only for the first four. For the rest, use just the exception object.

Also make `exceptionList` an OrderedDict and use the `partIndex` as a key and the traceback/exception as the value.

Rename `exceptionList` to `exceptions`. I don't like encoding the type in variable names.